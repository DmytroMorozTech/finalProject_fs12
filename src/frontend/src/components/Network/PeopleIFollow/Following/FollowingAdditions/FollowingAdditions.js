import React from 'react'
import FollowingAdditionsItem from './FollowingAdditionsItem/FollowingAdditionsItem'

function FollowingAdditions ({handleAll, handleConnections, handleOutOfNetwork, handleCompanies}) {
  const items = [
    {
      title: 'All',
      subtitle: '',
      onClick: handleAll
    },
    {
      title: 'Connections',
      subtitle: 'Connections you are following',
      onClick: handleConnections
    },
    {
      title: 'Out-of-Network',
      subtitle: 'People you are following, who are not connections',
      onClick: handleOutOfNetwork
    },
    {
      title: 'Companies',
      subtitle: 'Companies you are following',
      onClick: handleCompanies
    }
  ]

  return (
    <div>
      {items.map(({ title, ...rest }, i) => (
        <FollowingAdditionsItem key={i} title={title} {...rest}/>
      ))}
    </div>
  )
}

export default FollowingAdditions
