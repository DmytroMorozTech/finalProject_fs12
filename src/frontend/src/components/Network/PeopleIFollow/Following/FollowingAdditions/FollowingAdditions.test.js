import { shallow } from 'enzyme'
import React from 'react'
import FollowingAdditions from './FollowingAdditions'

const setUp = (props) => shallow(<FollowingAdditions {...props}/>)

describe('should render FollowingAdditions component', () => {
  let component
  let mockFunc = jest.fn()
  beforeEach(() => {
    component = setUp({
      handleAll: mockFunc,
      handleConnections: mockFunc,
      handleOutOfNetwork: mockFunc,
      handleCompanies: mockFunc
    })
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
