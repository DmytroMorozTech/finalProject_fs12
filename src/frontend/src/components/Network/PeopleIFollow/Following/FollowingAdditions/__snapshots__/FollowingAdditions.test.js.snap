// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FollowingAdditions component to match snapshot 1`] = `
<div>
  <FollowingAdditionsItem
    key="0"
    onClick={[MockFunction]}
    subtitle=""
    title="All"
  />
  <FollowingAdditionsItem
    key="1"
    onClick={[MockFunction]}
    subtitle="Connections you are following"
    title="Connections"
  />
  <FollowingAdditionsItem
    key="2"
    onClick={[MockFunction]}
    subtitle="People you are following, who are not connections"
    title="Out-of-Network"
  />
  <FollowingAdditionsItem
    key="3"
    onClick={[MockFunction]}
    subtitle="Companies you are following"
    title="Companies"
  />
</div>
`;
